hgmo.js

```diff
@@ -0,0 +1,59 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+"use strict";
+
+let filename = "";
+let revision = "";
+const linePattern = new RegExp("^l([0-9]+)$");
+
+document.querySelectorAll("title").forEach(title => {
+  const titlePattern = new RegExp("^mozilla-central: ([^@]+)@([0-9a-f]+)(?: \\(annotated\\))?$");
+  const m = title.innerText.match(titlePattern);
+  if (m) {
+    filename = m[1];
+    revision = m[2];
+  }
+  return;
+});
+
+async function applyOverlay(revPromise, path) {
+  let result = await getCoverage(revPromise, path);
+  if (!result.hasOwnProperty("data")) {
+    throw new Error("No 'data' field");
+  }
+  const data = result["data"];
+  document.querySelectorAll("[id^='l']").forEach(e => {
+    const m = e.id.match(linePattern);
+    if (m) {
```

Comment:
Nit: if you return early `if (!m)` you can avoid indentation in the following part